### DISCUSSION TITLE
Code of conduct

### DISCUSSION THREAD

**Discussion by pepe** [janet-lang/janet#558] [General]
In the https://github.com/janet-lang/janet/discussions/552 @bakpakin mentioned the Code of Conduct for the Janet language. 

I found Fennel CoC accessible and well written https://git.sr.ht/~technomancy/fennel/tree/main/CODE-OF-CONDUCT.md

Do you have any example of the CoC you like or something you would have changed in the Fennel one above?

**Comment by sogaiu** [comment]
I scanned the Fennel one and to be honest, I did not comprehend a fair bit of it -- there seems to be terminology I am not too familiar with and I wonder whether that wouldn't be the case for other folks too -- particularly people who are not from the U.S.

As a comparison, here is one from the Clojurian's Slack: https://github.com/clojurians/community-development/blob/master/Code-of-Conduct.md

This one I feel I can at least understand.

No real opinions on its content though.

**Comment by bakpakin** [comment]
Things in the Janet-sphere have thus far been fairly calm, but there have several small incidents that I have had to stop and I would like to not have to again.

* Conspiracy/right-wing conversation (or generally any political conversation)
* Sexual topics - in the interest of inclusiveness, I would also like to keep chat language to at most a PG-13 movie rating (American prudishness, I know). 
* Non-constructive criticism/trolls ("this is garbage!") - this is a sticky one for me, as I don't want to stifle useful criticism, even if it poorly worded or constructed. If people are frustrated, there is usually a reason, and if that reason is Janet I want to know. That said, we will never please everyone and ignorant and mean spirited criticism is draining.
* Off-topic stuff. Honestly I don't think this is a huge, but obviously we want to prevent people from hijacking our chat rooms for random things.
* Generally unpleasant people - this has been very rare, and usually these individual leave on their own, but whatever COC we adopt I hope it codifies that you can't be a jerk in chat. Often people aren't even trying to be mean, they are just not particularly socially skilled, but "negative vibes" really do make chat that much enjoyable for everyone else.

Anyways, most COC's I've read more or less cover this stuff, and I think almost any would be good, but going back to the source of the Fennel COC I think https://www.contributor-covenant.org/version/2/0/code_of_conduct/ is a good start.

The biggest question to me right now is enforcement. Right now, I am responsible for banning people and moderating chat, so naturally no one has been banned and I have just told people to stop talking about certain topics a few times. I have not been a very attentive moderator and I think we may want multiple moderators who can better cover the chat (and now GitHub discussions).

**Reply by sogaiu** [reply]
Thank you, I find this COC much easier to understand.  I will try to digest it.

(On the enforcement angle, at least for my off-topic comments, I will make an effort on that front to reduce.)

**Reply by sogaiu** [reply]
Reading through the [sample COC](https://www.contributor-covenant.org/version/2/0/code_of_conduct/), I see that "Bans" of certain sorts are mentioned.

Is this technically possible at this time for GH Discussions?

**Reply by pepe** [reply]
Thank you @bakpakin what you have written is very on point. A perfect building block. 

No conspiracy/right-wind is a must. Even with the nick I have, I do not stand any of it.

PEG-13 is an excellent threshold, and the inclusiveness is essential. That is why I have it in the first paragraph of the first article in my Óde to Janet.

For me, the hardest is the offtopic chats. As I am guilty of doing them from time to time, I also feel that they could be some community's glue. Still, I would try to limit them until we have some specialized "water cooler" channel.

**Reply by pepe** [reply]
I have read through the Contributor Covenant and found it just enough. I would take it as a start and then see what needs to be added and/or changed.

Moderation is always hard, but I think we have some good old members in all main broader timezones (UTC and +- 8 hours) and you can choose and trust some of them with this work. As long as you @bakpakin will stay as the sole judge on the verdict (especially temporary and permanent bans).

**Reply by sogaiu** [reply]
So I looked around a bit on the angle of "bans" for gh.

I guess may be these kinds can be used?

* https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/limiting-interactions-in-your-repository
* https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/blocking-a-user-from-your-organization

FWIW, I'm just looking into the technical feasibility of "encforcement" -- I don't really yet have an opinion on specifics of what should be in the COC, nor how it would get "encforced".

**Comment by pepe** [comment]
I just read https://github.com/ziglang/zig/blob/master/CODE_OF_CONDUCT.md, and the one thing I like apart from its simplicity is that moderators are named in the document.
